### PR TITLE
feat:add dev check

### PR DESCRIPTION
**Pull request by GrapeBaBa**
Add a basic ruff check for the project and make github CI works.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        branches: [ "master" ]
+    pull_request:
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.10", "3.11"]
```
Comment (norswap) [inline .github/workflows/build.yml:14]: Is there a specific reason to try those two versions instead of just one?

File: README.md [hunk 2]
```diff
@@ -20,7 +20,7 @@ you can automatically subsidize gas for transactions that match certain criteria
 
 ## Prerequisites
 
-- Python 3 (to run the `roll.py` script)
+- Python >= 3.10 (to run the `roll.py` script)
```
Comment (norswap) [inline README.md:23]: Do we know if we have dependencies that require python >= 3.10?

**Review by norswap — APPROVED** [review]
Looks solid, asked a few questions about Python versions.

File: README.md (hunk 2 above)
Comment (GrapeBaBa) [reply README.md:23]: Yes, https://github.com/0xFableOrg/simple-op-stack-rollup/blob/master/libroll.py#L8 this syntax need >=3.10

**Review by ayazabbas — APPROVED** [review]

**Pull request merged**